apps/web/components/NavigationBar/HideOnScroll.tsx

```diff
@@ -0,0 +1,25 @@
+"use client"
+
+import { useEffect, useState } from "react"
+import { cn } from "utils/cn"
+
+export function HideOnScroll({ children }) {
+  const [lastScrollTop, setLastScrollTop] = useState(0)
+  const [hidden, setHidden] = useState(false)
+
+  useEffect(() => {
+    function handleScroll() {
+      const currentScrollTop = window.scrollY || document.documentElement.scrollTop
+      if (currentScrollTop > lastScrollTop && currentScrollTop > 100) {
```

Comment:
`currentScrollTop > lastScrollTop && currentScrollTop > 100` could be assigned to a variable e.g. `isHidden` and then used in setHidden without IFs